controllers/capabilities/developeruser_controller.go

```diff
@@ -208,6 +264,37 @@ func (r *DeveloperUserReconciler) findParentAccount(userCR *capabilitiesv1beta1.
 	return devAccountCR, nil
 }
 
+func (r *DeveloperUserReconciler) removeDeveloperUserFrom3scale(developerUser *capabilitiesv1beta1.DeveloperUser) error {
+	logger := r.Logger().WithValues("developerUser", client.ObjectKey{Name: developerUser.Name, Namespace: developerUser.Namespace})
+
+	// Attempt to remove developerUser only if developerUser.Status.ID is present
+	if developerUser.Status.ID == nil {
+		logger.Info("could not remove developerUser because ID is missing in status")
+		return nil
+	}
```

Comment:
Should `developerUser.Status.AccountID` be checked also or can we assume `AccountID` is not `nil` when `ID` is not `nil`  ?

Reply:
I think it should be checked as well. If any field is missing, the operator should consider to create a new one.